### PR TITLE
LibJS: Implement Temporal.PlainTime.prototype.round

### PR DESCRIPTION
**Pull request by Lubrsi**
Ticks one checkbox in #8982.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainTimePrototype.cpp [hunk 1]
```diff
@@ -267,6 +268,65 @@ JS_DEFINE_NATIVE_FUNCTION(PlainTimePrototype::with)
...
+    auto* options = TRY(get_options_object(global_object, vm.argument(0)));
+
+    // 5. Let smallestUnit be ? ToSmallestTemporalUnit(options, « "year", "month", "week", "day" », undefined).
+    auto smallest_unit_value = TRY(to_smallest_temporal_unit(global_object, *options, { "year"sv, "month"sv, "week"sv, "day"sv }, {}));
+
+    // 6. If smallestUnit is undefined, throw a RangeError exception.
+    if (!smallest_unit_value.has_value())
+        return vm.throw_completion<RangeError>(global_object, ErrorType::OptionIsNotValidValue, vm.names.undefined.as_string(), "smallestUnit");
+
+    // At this point smallest_unit_value can only be a string
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/PlainTimePrototype.cpp:294]: nitpick: when we add internal comments inside functions that are spec-commented we usually prefix them with `NOTE: `

File: Userland/Libraries/LibJS/Tests/builtins/Temporal/PlainTime/PlainTime.prototype.round.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            const plainTime = new Temporal.PlainTime(18, 15, 9, 100, 200, 300);
+            plainTime.round();
+        }).toThrowWithMessage(TypeError, "Required options object is missing or undefined");
+    });
+
+    test("invalid rounding mode", () => {
+        expect(() => {
+            const plainTime = new Temporal.PlainTime(18, 15, 9, 100, 200, 300);
+            plainTime.round({ smallestUnit: "second", roundingMode: "serenityOS" });
+        }).toThrowWithMessage(RangeError, "is not a valid value for option roundingMode");
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/Temporal/PlainTime/PlainTime.prototype.round.js:84]: Please don't rely on "message string but it's actually a regex" for new tests. Just provide the full expected error message.

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainTimePrototype.cpp [hunk 3]
```diff
@@ -267,6 +268,65 @@ JS_DEFINE_NATIVE_FUNCTION(PlainTimePrototype::with)
...
+    auto rounding_mode = TRY(to_temporal_rounding_mode(global_object, *options, "halfExpand"));
+
+    double maximum;
+
+    // 8. If smallestUnit is "hour", then
+    if (smallest_unit == "hour"sv) {
+        // a. Let maximum be 24.
+        maximum = 24;
+    }
+
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/PlainTimePrototype.cpp:307]: nit: for the sake of CxByte's sanity we have opted for omitting the empty lines between multi-step if else :^)

**Comment by Lubrsi** [comment]
Done :+1:

File: Userland/Libraries/LibJS/Tests/builtins/Temporal/PlainTime/PlainTime.prototype.round.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    });
+
+    test("increment may not be NaN", () => {
+        expect(() => {
+            const plainTime = new Temporal.PlainTime(18, 15, 9, 100, 200, 300);
+            plainTime.round({ smallestUnit: "second", roundingIncrement: NaN });
+        }).toThrowWithMessage(RangeError, "NaN is not a valid value for option roundingIncrement");
+    });
+
+    test("increment may smaller than 1 or larger than maximum", () => {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/Temporal/PlainTime/PlainTime.prototype.round.js:107]:
```suggestion
    test("increment may not be smaller than 1 or larger than maximum", () => {
```

**Review by linusg — COMMENTED** [review]
sorry, one more that I missed earlier...

File: Userland/Libraries/LibJS/Tests/builtins/Temporal/PlainTime/PlainTime.prototype.round.js (hunk 4 above)
Comment (Lubrsi) [reply Userland/Libraries/LibJS/Tests/builtins/Temporal/PlainTime/PlainTime.prototype.round.js:107]: Done :+1:

**Review by linusg — APPROVED** [review]

**Pull request merged**